meeting-notes/2022-05-14.md

```diff
@@ -0,0 +1,261 @@
+# 5/14/22
+
+@pcohen, @rntz
+
+## TODOs
+
+- [ ] Phil JSON:
+  - add cursor selection ranges to JSON output
+- [ ] rntz JSON:
+  - Add selections (including the anchor)
+  - Multiple cursors
+  - Buffer filename should be null if nonexistent
+  - camelCase
+
+@rntz sez: "maybe we should put all these files in a ~/.cursorless/ directory?"
+
+## Notes
+
+**what phil's fork of the cursorless vscode plugin does**
+
+1. writes hats to a file when there's a change to those hats
+
+**phil next steps**
+
+- Migrate "watch JB/Emacs editor state and tell VS Code to navigate to that same
+  location (thus updating the hats)"logic to cursorless fork
+  (file watcher)
+
+- Make new JetBrains plugin also publish standardized \~/.editor-state.json,
+  since it\'s current JSON is overly complicated for this use case
+
+- Export forked changes in knausj VSCode command server
+
+- Export changes from custom command server
+
+**rntz next steps**
+
+- ~~Install and get building (./build)~~
+
+- Get emacs reading \~/.vscode-hats.json and applying hats
+
+- Think about how commands will be sent to visual studio code\
+  *issue: *vscode "command server"waits for a keystroke to tell it to read a
+  file talon writes to execute cursorless commands. just use a file watcher?
```

Comment:
you could, but this would lead to race conditions with partial writes.  VSCode command server relies on the fact that the keystroke is not issued until write is complete.  Fwiw Talon side (command client) [polls](https://github.com/pokey/talon-vscode-command-client/blob/1f5b37394c21e015a3d92dfbb804903d1d142cf2/command_client.py#L251-L287), waiting for a final newline.  I guess you could do the same?

Reply:
Couldn't we just write atomically?

Reply:
🤷. I believe certain platforms, at certain layers, with certain sized writes, guarantee atomicity, but no idea whether you can reliably guarantee it in a cross-platform fashion